src/Pim/Bundle/CatalogBundle/Elasticsearch/ProductIndexer.php

```diff
@@ -64,6 +66,33 @@ public function indexAll(array $products, array $options = [])
         }
 
         $this->indexer->bulkIndexes($this->indexType, $normalizedProducts, 'id');
+
+        if (!isset($options['associated_products']) || false === $options['associated_products']) {
```

Comment:
Maybe the naming is not clear enough. The idea is to not fall in an infinite loop of indexation.
First time `indexAll` is called (from the subscriber), the option is not set, so we can call the method `indexAssociatedProducts`. This method will create the list of associated product to index and call `indexAll` again, but sets the option to `true`, so `indexAssociatedProducts` is not called again.